### PR TITLE
Spawn player and update time

### PR DESCRIPTION
**Pull request by Thomas-Harit**

### CODE REVIEW

File: cubic-server/Default/DefaultWorld.cpp [hunk 1]
```diff
@@ -23,3 +26,15 @@ void DefaultWorld::initialize()
...
+void DefaultWorld::updateTime() {
+    std::shared_ptr<std::vector<uint8_t>> data;
+    // add tick to age of the world and time
+    _age += 1;
+    _time += 1;
+    if (_time > 24000)
+        _time = 0;
+
+    // send packets to clients (missing clients in architecture)
+    data = protocol::createUpdateTime({_age, _time});
```
Comment (huntears) [inline cubic-server/Default/DefaultWorld.cpp:39]: The clients will be added by #48 , so maybe we can wait for that to be merged before merging this

File: cubic-server/Dimension.cpp [hunk 2]
```diff
@@ -11,3 +13,32 @@ void Dimension::initialize()
...
 }
+
+// should be called at every movement for better efficiency
+void Dimension::spawnPlayer()
+{
+    long distance;
+    std::shared_ptr<std::vector<uint8_t>> data;
+
+    // scan every player to each other if entering render distance
+    /*for (auto player = _players; player < _players.end(); player++) {
```
Comment (huntears) [inline cubic-server/Dimension.cpp:24]: Same here, maybe waiting for #48 might be a good idea

File: protocol/typeSerialization.hpp [hunk 3]
```diff
@@ -281,6 +281,13 @@ namespace protocol
         return (a << 64) | b;
     }
 
+    constexpr void addUUID(std::vector<uint8_t> &out, const __int128 &data)
+    {
+        for (uint8_t shift_value = 120; shift_value >= 0; shift_value -= 8) {
+            out.push_back((data >> shift_value) && 0xFF);
+        }
+    }
```
Comment (huntears) [inline protocol/typeSerialization.hpp:289]: Why not just use addLong two times ?
#48 also adds this function so once again, I think waiting for merge is the good move.

File: cubic-server/Default/DefaultWorld.hpp [hunk 4]
```diff
@@ -12,10 +12,14 @@ class DefaultWorld : public World
     DefaultWorld();
     void tick() override;
     void initialize() override;
+    void updateTime();
```
[the remark this answers is not in the document]
Comment (huntears) [reply cubic-server/Default/DefaultWorld.hpp:15]: Looking at wiki.vg it looks like I was wrong and the time is indeed per world, not per dimensions

**Review by huntears — DISMISSED** [review]

**Review by huntears — DISMISSED** [review]

**Review by STMiki — DISMISSED** [review]

**Review by huntears — DISMISSED** [review]

**Review by huntears — APPROVED** [review]

**Review by STMiki — APPROVED** [review]

**Pull request merged**